ee/tabby-webserver/src/schema/auth.rs

```diff
@@ -143,11 +201,17 @@ pub trait AuthenticationService: Send + Sync {
         password1: String,
         password2: String,
         invitation_code: Option<String>,
-    ) -> FieldResult<RegisterResponse>;
-    async fn token_auth(&self, email: String, password: String) -> FieldResult<TokenAuthResponse>;
-    async fn refresh_token(&self, refresh_token: String) -> FieldResult<RefreshTokenResponse>;
-    async fn verify_token(&self, access_token: String) -> FieldResult<VerifyTokenResponse>;
-    async fn is_admin_initialized(&self) -> FieldResult<bool>;
+    ) -> std::result::Result<RegisterResponse, RegisterError>;
+
+    async fn token_auth(
+        &self,
+        email: String,
+        password: String,
+    ) -> std::result::Result<TokenAuthResponse, TokenAuthError>;
+
+    async fn refresh_token(&self, refresh_token: String) -> Result<RefreshTokenResponse>;
+    async fn verify_token(&self, access_token: String) -> Result<VerifyTokenResponse>;
+    async fn is_admin_initialized(&self) -> Result<bool>;
```

Comment:
rule of thumb:
1. By default, starts with anyhow::Result.
2. If there's a need for Validation / Other error type, define a new error and iterate.

Sounds like a good iterative approach?

cc @darknight